py-polars/polars/internals/dataframe/frame.py

```diff
@@ -6468,15 +6477,19 @@ def rows(self, named: bool = False) -> list[tuple[Any, ...]] | list[Any]:
             return self._df.row_tuples()
 
     @overload
-    def iterrows(self, named: Literal[False] = ...) -> Iterator[tuple[Any, ...]]:
+    def iterrows(
+        self, named: Literal[False] = ..., buffer_size: int = ...
+    ) -> Iterator[tuple[Any, ...]]:
         ...
 
     @overload
-    def iterrows(self, named: Literal[True] = ...) -> Iterator[Any]:
+    def iterrows(
+        self, named: Literal[True] = ..., buffer_size: int = ...
+    ) -> Iterator[Any]:
         ...
 
     def iterrows(
-        self, named: bool = False
+        self, named: bool = False, buffer_size: int = 500
```

Comment:
Maybe we can make `buffer_size` a keyword-only argument? That would make it easier to handle changes later if we manage to further optimize the iterrows process.

```suggestion
        self, named: bool = False, *, buffer_size: int = 500
```

Reply:
I've got a few other ideas/improvements for this method - can make it a kwarg-only param alongside the other updates (maybe tonight/tomorrow) 👍